### ISSUE TITLE
alignment of = sign in enum typedef

### ISSUE THREAD

**Issue by stleusc** [uncrustify/uncrustify#1030]
I'm sure I saw an option that produced the following output. Can't find it anymore :-(

```C
typedef enum
{
    XXXXXXXXXXXXXXXXX      = 0U,
    ZZZZZZZZZZZ            = 1U,
    TTTTTTTT               = 2U
} MyType;
```

Any advice?

**Comment by stleusc** [comment]
I was playing with

align_enum_equ_span=4
align_enum_equ_thresh=4

and different values, but no alignment at all. Maybe a global overall switch missing?

**Comment by stleusc** [comment]
playing with

align_assign_span=1
align_assign_thresh=1

turns on the above formatting but also aligns = in normal code.

There has to be a way to turn it ON in enum but OFF in code, why else have these options?

**Comment by CDanU** [comment]
Please actually read the [ISSUE_TEMPLATE](https://github.com/uncrustify/uncrustify/blob/master/.github/ISSUE_TEMPLATE)
Additionally to your expected result I also need your input.

**Comment by stleusc** [comment]
I'll see that I update the description better.

Example input:

```C
typedef enum
{
    XXXXXXXXXXXXXXXXX = 0U,
    ZZZZZZZZZZZ = 1U,
    TTTTTTTT = 2U
} MyType;
```

I understand the need for some global flag but then there should be a specific flag for EACH suboption.

Right now I don't see a way how I can align enum and NOT align code.

Best result is achieved so far using

```
align_assign_span=1
align_assign_thresh=1
align_enum_equ_span=1
align_enum_equ_thresh=0
```

But it will still align this code as well

```C
x = 3;
xy = 4;
```

will become

```C
x  = 3;
xy = 4;
```

[uncrustify.cfg](http://pastebin.com/Gk3qvjnH)

**Comment by CDanU** [comment]
Yes I agree that we should need more special options for the align_assign functionality which can also operate on their own. 

I tried to achieve proper results with this little prototype commit here: 
https://github.com/CDanU/uncrustify/commit/aa6e24811a84ea2a7ab15032fba69dae1c7110d9

That one breaks some test cases. I would have to understand what the AlignStack class is actually doing but I currently don't have the time for that.

**Comment by ostraat** [comment]
This issue is fixed in PR #1055.  Please take a look and comment.

**Comment by guy-maurel** [comment]
@ostraat Could you create a new PR with it?
